core/src/routes.rs

```diff
@@ -1,395 +1,502 @@
+use anyhow::{anyhow, Result};
+use cid::{Cid, CidGeneric};
+use iroh::PublicKey;
+use iroh_blobs::BlobFormat;
 use serde::{Deserialize, Serialize};
-use serde_json::Value;
-use utoipa::ToSchema;
+use time::OffsetDateTime as DateTime;
+use uuid::Uuid;
+
+use crate::{context::Signer, crp::ProviderType};
 
-/// A route defining a method for resolving a CID to its content and/or metadata associated with its content.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct Route {
-    /// CID Route Provider ID.
-    /// This optional value is only meant for use by CID Routers with multiple CID Route Providers.
-    /// CID Route providers themselves don't need to set it.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub crp_id: Option<String>,
-    /// Type of the route.
-    #[serde(rename = "type")]
-    pub type_: String,
-    /// Method for resolving a CID.
-    /// Schema for the `method` is defined by the `type` field.
-    pub method: Value,
-    /// Metadata for the route.
-    /// Schema for the `metadata` is defined by the `type` field.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub metadata: Option<Value>,
+    pub id: Uuid,
+    #[serde(with = "time::serde::rfc3339")]
+    pub created_at: DateTime,
+    #[serde(with = "time::serde::rfc3339")]
+    pub verified_at: DateTime,
+    pub provider: ProviderType,
+    pub cid: CidGeneric<64>,
+    pub blob_format: BlobFormat,
```

Comment:
I think we can remove `BlobFormat` and just rely on the CID's multicodec to track "raw" vs "blake-hashseq". This also leaves it more open to track other possible enums that don't fall under `BlobFormarawt` like if we want to indicate raw git objects for git commits or something.

Reply:
While working through this I've come to think we should keep it for a single reason: it's far more useful as a selection method within the database than trying to parse through the CID bits in sqlite.

Reply:
That makes sense, in that case can we generalize it to `multicodec` and that can be used in the future if there are more than 2 codecs that end being used, and if not, not harm no foul